src/server/common/model/destination/destination.js

```diff
@@ -0,0 +1,36 @@
+import Joi from 'joi'
+import { Model, validateAgainstSchema } from '../model.js'
+
+const validationSchema = Joi.object().required()
+
+/**
+ * @import {RawPayload} from '../model.js'
+ */
+
+export class Destination extends Model {
```

Comment:
I'd be tempted to make a SectionModel that only has the interface we need for sections. 

At this point in time I think it just needs `fromState` and `validate`?